spec/blocksync/verification.md

```diff
@@ -0,0 +1,120 @@
+## Block verification
+
+When blocksyncing, a node is not participating in conensus. It is receiving blocks that have already been decided and committed. To avoid being fooled by malicious peers, the node has to verify the received blocks before executing the transactions and storing the block in its store. 
+
+
+The verification in blocksync aims to apply the same logic as the [light client verification](../light-client/verification/README.md). The safety guarantees of the light client model are provided by verifiying blocks starting from an **initial trusted state** and using validators who were bonded within a **trusting period**. Once this period expires, the validators could have unbonded, and we cannot rely on them being hoenst anymore. In blocksync, we can be **outside** the trusting period. Therefore, the guarantees provided by blocksync verification can be divided in two groups: 
+
+- The block is within the trusting period, we can punish validators, and provide guarantees that if a block is signed by trusted validators, they are indeed correct. 
+- The block is outside the trusting period and we cannot guarantee that the validators are still honest. We can however make a potential attack by those validators more complicated by additionally verifying the block using other peers as *witnesses*.
+
+A state becomes trusted once it passes the validation. The section below discusses this state in more details.
+
+### Trusted state
+
+Currently there are two possible ways to obtain a trusted state :
+ 1. If the node is blocksync-ing from genesis, we have to verify the very first block and mark it as the initial trusted state.
+ 
+ [//]: # (trust the validators provided in the initial state and use them to verify the initial block received from peers.)
+
+ 2. The node loads the last block stored in its local block store. 
+
+**Case 1 - blocksync from genesis**
+
+In this case we only verify that the validator hash of the block matches the validator set of the initial state. We will use the initially provided validator set for verification and further verify the block against witne+sses. In this scenario, this state is very likely to be outside the trusting period. We will accept this block as a trusted state and store it inside the node's block store. 
+
+It is worth noting that, running block sync from the first height is significantly slower than running statesync first. However, statesync does not keep the entire blockchain history and some operators might opt not to state sync. The reason is that, if sufficiently many nodes state sync and other nodes who have historical data fail or leave the network, we have no history.
```

Comment:
> we have no history.

I'd rather say something like "the network ends up with gaps in the block history"